From 9ea9caf81ddbd2c3822079e2dd77268733c85ce0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Sat, 28 Apr 2018 14:50:11 +0100
Subject: [PATCH 03/45] gst-libs: include config.h in all source files

This will be needed later when we get our export define from config.h

(cherry picked from commit ca15315565912172b2dec5c3e7887e34dc4f8ccf)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst-libs/gst/audio/audio-quantize.c                   | 3 +++
 gst-libs/gst/audio/gstaudiobasesink.c                 | 3 +++
 gst-libs/gst/audio/gstaudiometa.c                     | 3 +++
 gst-libs/gst/audio/gstaudioringbuffer.c               | 3 +++
 gst-libs/gst/audio/gstaudiosink.c                     | 3 +++
 gst-libs/gst/audio/gstaudiosrc.c                      | 3 +++
 gst-libs/gst/gl/egl/gstegl.c                          | 3 +++
 gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c             | 3 +++
 gst-libs/gst/pbutils/gstpluginsbaseversion.c          | 3 +++
 gst-libs/gst/rtp/gstrtcpbuffer.c                      | 3 +++
 gst-libs/gst/rtp/gstrtpbasedepayload.c                | 3 +++
 gst-libs/gst/rtp/gstrtpbuffer.c                       | 3 +++
 gst-libs/gst/rtp/gstrtphdrext.c                       | 3 +++
 gst-libs/gst/rtp/gstrtppayloads.c                     | 3 +++
 gst-libs/gst/rtsp/gstrtspmessage.c                    | 3 +++
 gst-libs/gst/rtsp/gstrtsprange.c                      | 3 +++
 gst-libs/gst/rtsp/gstrtsptransport.c                  | 3 +++
 gst-libs/gst/rtsp/gstrtspurl.c                        | 3 +++
 gst-libs/gst/sdp/gstmikey.c                           | 3 +++
 gst-libs/gst/tag/gsttageditingprivate.c               | 3 +++
 gst-libs/gst/video/convertframe.c                     | 3 +++
 gst-libs/gst/video/gstvideoaffinetransformationmeta.c | 3 +++
 gst-libs/gst/video/gstvideometa.c                     | 3 +++
 gst-libs/gst/video/gstvideopool.c                     | 3 +++
 gst-libs/gst/video/gstvideotimecode.c                 | 3 +++
 gst-libs/gst/video/video-dither.c                     | 3 +++
 gst-libs/gst/video/video-event.c                      | 3 +++
 gst-libs/gst/video/video-tile.c                       | 3 +++
 28 files changed, 84 insertions(+)

diff --git a/gst-libs/gst/audio/audio-quantize.c b/gst-libs/gst/audio/audio-quantize.c
index 5ac54609e0..8c627173bc 100644
--- a/gst-libs/gst/audio/audio-quantize.c
+++ b/gst-libs/gst/audio/audio-quantize.c
@@ -25,6 +25,9 @@
  *         generated by dmaker
  *         http://shibatch.sf.net
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <gst/gst.h>
 #include <string.h>
diff --git a/gst-libs/gst/audio/gstaudiobasesink.c b/gst-libs/gst/audio/gstaudiobasesink.c
index ff4e4e2223..1256e9ce9f 100644
--- a/gst-libs/gst/audio/gstaudiobasesink.c
+++ b/gst-libs/gst/audio/gstaudiobasesink.c
@@ -30,6 +30,9 @@
  * ::create_ringbuffer vmethod. This base class will then take care of
  * writing samples to the ringbuffer, synchronisation, clipping and flushing.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/audio/gstaudiometa.c b/gst-libs/gst/audio/gstaudiometa.c
index e7da037557..83b901873c 100644
--- a/gst-libs/gst/audio/gstaudiometa.c
+++ b/gst-libs/gst/audio/gstaudiometa.c
@@ -26,6 +26,9 @@
  * audio buffers. These functions in this module help to create and attach the
  * meta as well as extracting it.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/audio/gstaudioringbuffer.c b/gst-libs/gst/audio/gstaudioringbuffer.c
index 991b774f3a..90d0cac006 100644
--- a/gst-libs/gst/audio/gstaudioringbuffer.c
+++ b/gst-libs/gst/audio/gstaudioringbuffer.c
@@ -33,6 +33,9 @@
  * implementations.
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/audio/gstaudiosink.c b/gst-libs/gst/audio/gstaudiosink.c
index cc7b791bad..4325f71a30 100644
--- a/gst-libs/gst/audio/gstaudiosink.c
+++ b/gst-libs/gst/audio/gstaudiosink.c
@@ -48,6 +48,9 @@
  * together with #GstAudioBaseSink using a default implementation of a
  * #GstAudioRingBuffer that uses threads.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/audio/gstaudiosrc.c b/gst-libs/gst/audio/gstaudiosrc.c
index 008921fa91..16b090c209 100644
--- a/gst-libs/gst/audio/gstaudiosrc.c
+++ b/gst-libs/gst/audio/gstaudiosrc.c
@@ -41,6 +41,9 @@
  * together with #GstAudioBaseSrc using a default implementation of a
  * #GstAudioRingBuffer that uses threads.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/gl/egl/gstegl.c b/gst-libs/gst/gl/egl/gstegl.c
index 9f455e52cd..9d7b83f400 100644
--- a/gst-libs/gst/gl/egl/gstegl.c
+++ b/gst-libs/gst/gl/egl/gstegl.c
@@ -17,6 +17,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <gst/gl/egl/gstegl.h>
 
diff --git a/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c b/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c
index 0d68b92514..ec24ef3bad 100644
--- a/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c
+++ b/gst-libs/gst/gl/gbm/gstglwindow_gbm_egl.c
@@ -17,6 +17,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <poll.h>
 
diff --git a/gst-libs/gst/pbutils/gstpluginsbaseversion.c b/gst-libs/gst/pbutils/gstpluginsbaseversion.c
index 3b6f9ea749..d4075b203d 100644
--- a/gst-libs/gst/pbutils/gstpluginsbaseversion.c
+++ b/gst-libs/gst/pbutils/gstpluginsbaseversion.c
@@ -29,6 +29,9 @@
  *
  * The version macros get defined by including &lt;gst/pbutils/pbutils.h&gt;.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gstpluginsbaseversion.h"
 
diff --git a/gst-libs/gst/rtp/gstrtcpbuffer.c b/gst-libs/gst/rtp/gstrtcpbuffer.c
index 5a858c75e7..ce0b91f829 100644
--- a/gst-libs/gst/rtp/gstrtcpbuffer.c
+++ b/gst-libs/gst/rtp/gstrtcpbuffer.c
@@ -38,6 +38,9 @@
  * gst_rtcp_packet_move_to_next().
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/rtp/gstrtpbasedepayload.c b/gst-libs/gst/rtp/gstrtpbasedepayload.c
index 8773cee294..2a9259a4b8 100644
--- a/gst-libs/gst/rtp/gstrtpbasedepayload.c
+++ b/gst-libs/gst/rtp/gstrtpbasedepayload.c
@@ -25,6 +25,9 @@
  *
  * Provides a base class for RTP depayloaders
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gstrtpbasedepayload.h"
 
diff --git a/gst-libs/gst/rtp/gstrtpbuffer.c b/gst-libs/gst/rtp/gstrtpbuffer.c
index 17683922c8..fb4f573ba7 100644
--- a/gst-libs/gst/rtp/gstrtpbuffer.c
+++ b/gst-libs/gst/rtp/gstrtpbuffer.c
@@ -29,6 +29,9 @@
  * 'application/x-rtp' #GstCaps.
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gstrtpbuffer.h"
 
diff --git a/gst-libs/gst/rtp/gstrtphdrext.c b/gst-libs/gst/rtp/gstrtphdrext.c
index 5d483c1478..f653f2f3df 100644
--- a/gst-libs/gst/rtp/gstrtphdrext.c
+++ b/gst-libs/gst/rtp/gstrtphdrext.c
@@ -24,6 +24,9 @@
  * @see_also: #GstRTPBasePayload, #GstRTPBaseDepayload, gstrtpbuffer
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gstrtphdrext.h"
 
diff --git a/gst-libs/gst/rtp/gstrtppayloads.c b/gst-libs/gst/rtp/gstrtppayloads.c
index 702cceb20f..2c41782044 100644
--- a/gst-libs/gst/rtp/gstrtppayloads.c
+++ b/gst-libs/gst/rtp/gstrtppayloads.c
@@ -31,6 +31,9 @@
  * and get session bandwidth information.
  *
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/rtsp/gstrtspmessage.c b/gst-libs/gst/rtsp/gstrtspmessage.c
index 67bdd49016..fb4b65161f 100644
--- a/gst-libs/gst/rtsp/gstrtspmessage.c
+++ b/gst-libs/gst/rtsp/gstrtspmessage.c
@@ -50,6 +50,9 @@
  *
  * Provides methods for creating and parsing request, response and data messages.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/rtsp/gstrtsprange.c b/gst-libs/gst/rtsp/gstrtsprange.c
index 624fc71944..ef41c8ae78 100644
--- a/gst-libs/gst/rtsp/gstrtsprange.c
+++ b/gst-libs/gst/rtsp/gstrtsprange.c
@@ -47,6 +47,9 @@
  *
  * Provides helper functions to deal with time ranges.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <math.h>
 #include <stdio.h>
diff --git a/gst-libs/gst/rtsp/gstrtsptransport.c b/gst-libs/gst/rtsp/gstrtsptransport.c
index 8976ba4a96..51be5f9ec0 100644
--- a/gst-libs/gst/rtsp/gstrtsptransport.c
+++ b/gst-libs/gst/rtsp/gstrtsptransport.c
@@ -48,6 +48,9 @@
  *
  * Provides helper functions to deal with RTSP transport strings.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 #include <stdlib.h>
diff --git a/gst-libs/gst/rtsp/gstrtspurl.c b/gst-libs/gst/rtsp/gstrtspurl.c
index 9ae15ca8c1..dfb76df1db 100644
--- a/gst-libs/gst/rtsp/gstrtspurl.c
+++ b/gst-libs/gst/rtsp/gstrtspurl.c
@@ -47,6 +47,9 @@
  *
  * Provides helper functions to handle RTSP urls.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <stdlib.h>
 #include <string.h>
diff --git a/gst-libs/gst/sdp/gstmikey.c b/gst-libs/gst/sdp/gstmikey.c
index 3e999512d9..7117f9853f 100644
--- a/gst-libs/gst/sdp/gstmikey.c
+++ b/gst-libs/gst/sdp/gstmikey.c
@@ -29,6 +29,9 @@
  *
  * Since: 1.4
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/tag/gsttageditingprivate.c b/gst-libs/gst/tag/gsttageditingprivate.c
index 0d18e89049..db531f6d50 100644
--- a/gst-libs/gst/tag/gsttageditingprivate.c
+++ b/gst-libs/gst/tag/gsttageditingprivate.c
@@ -16,6 +16,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gsttageditingprivate.h"
 
diff --git a/gst-libs/gst/video/convertframe.c b/gst-libs/gst/video/convertframe.c
index 61e9c1a8b1..faaba3ade7 100644
--- a/gst-libs/gst/video/convertframe.c
+++ b/gst-libs/gst/video/convertframe.c
@@ -18,6 +18,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 #include "video.h"
diff --git a/gst-libs/gst/video/gstvideoaffinetransformationmeta.c b/gst-libs/gst/video/gstvideoaffinetransformationmeta.c
index 083d1c4f69..139804425a 100644
--- a/gst-libs/gst/video/gstvideoaffinetransformationmeta.c
+++ b/gst-libs/gst/video/gstvideoaffinetransformationmeta.c
@@ -18,6 +18,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gstvideoaffinetransformationmeta.h"
 
diff --git a/gst-libs/gst/video/gstvideometa.c b/gst-libs/gst/video/gstvideometa.c
index be833747a3..0faf9b44c3 100644
--- a/gst-libs/gst/video/gstvideometa.c
+++ b/gst-libs/gst/video/gstvideometa.c
@@ -16,6 +16,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gstvideometa.h"
 
diff --git a/gst-libs/gst/video/gstvideopool.c b/gst-libs/gst/video/gstvideopool.c
index b6d2dea2cc..d02bc54b1b 100644
--- a/gst-libs/gst/video/gstvideopool.c
+++ b/gst-libs/gst/video/gstvideopool.c
@@ -16,6 +16,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "gst/video/gstvideometa.h"
 #include "gst/video/gstvideopool.h"
diff --git a/gst-libs/gst/video/gstvideotimecode.c b/gst-libs/gst/video/gstvideotimecode.c
index e7955782be..3dde19025e 100644
--- a/gst-libs/gst/video/gstvideotimecode.c
+++ b/gst-libs/gst/video/gstvideotimecode.c
@@ -16,6 +16,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <stdio.h>
 #include "gstvideotimecode.h"
diff --git a/gst-libs/gst/video/video-dither.c b/gst-libs/gst/video/video-dither.c
index 3f5e6bd723..3f244a45a1 100644
--- a/gst-libs/gst/video/video-dither.c
+++ b/gst-libs/gst/video/video-dither.c
@@ -16,6 +16,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <string.h>
 
diff --git a/gst-libs/gst/video/video-event.c b/gst-libs/gst/video/video-event.c
index 0cbd40061c..4533703071 100644
--- a/gst-libs/gst/video/video-event.c
+++ b/gst-libs/gst/video/video-event.c
@@ -18,6 +18,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include "video-event.h"
 
diff --git a/gst-libs/gst/video/video-tile.c b/gst-libs/gst/video/video-tile.c
index dd40ee6205..e22a24d9df 100644
--- a/gst-libs/gst/video/video-tile.c
+++ b/gst-libs/gst/video/video-tile.c
@@ -16,6 +16,9 @@
  * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
  * Boston, MA 02110-1301, USA.
  */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
 
 #include <gst/video/video-tile.h>
 
-- 
2.11.0

